src/App.jsx

```diff
@@ -14,45 +17,100 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_ALPHABETICALLY = 'alphabetically';
+const SORT_LENGTH = 'length';
+
+function getPreparedGood(goods, { sortField, isReverse }) {
+  const prepeardeGoods = [...goods];
+
+  if (sortField) {
+    prepeardeGoods.sort((item1, item2) => {
+      switch (sortField) {
+        case SORT_ALPHABETICALLY:
+          return item1.localeCompare(item2);
+        case SORT_LENGTH:
+          return item1.length - item2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReverse) {
+    prepeardeGoods.reverse();
+  }
+
+  return prepeardeGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReverse, setIsReverse] = useState(false);
+  const visibleGoods = getPreparedGood(goodsFromServer,
+    { sortField, isReverse });
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn(
+            'button is-info',
+            {
+              'is-light': sortField !== SORT_ALPHABETICALLY,
+            },
+          )}
+          onClick={() => setSortField(SORT_ALPHABETICALLY)}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={cn(
+            'button is-success',
+            {
+              'is-light': sortField !== SORT_LENGTH,
+            },
+          )}
+          onClick={() => setSortField(SORT_LENGTH)}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={cn(
+            'button is-warning',
+            {
+              'is-light': !isReverse,
+            },
+          )}
+          onClick={() => setIsReverse(!isReverse)}
```

Comment:
1. move to own handler function
2. it's not a good practice to pass getter into a setter. Use prev state

```
const handleReverseClick = () => {
  setIsReversed(prev => !prev)
}
```

```suggestion
          onClick={handleReverseClick}
```